### PR TITLE
Add multiple species

### PR DESCRIPTION
**Pull request by lvnhmd**

### CODE REVIEW

File: pages/common/views/components/add-another.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const visible = Array(this.props.totalCount).fill(false);
+    visible.fill(true, 0, this.props.visibleCount);
+    this.setState({ visible });
+  }
+
+  addAnother(e) {
+    e.preventDefault();
+    const visible = [ ...this.state.visible ];
+    visible[visible.lastIndexOf(true) + 1] = true;
+    this.setState({ visible });
```
Comment (joefitter) [inline pages/common/views/components/add-another.jsx]: you'll want to `return false` here if you don't want the `#` to end up in the URL

File: pages/pil/training/routers/modules.js [hunk 2]
```diff
@@ -14,13 +14,35 @@ module.exports = settings => {
...
   app.post('/', (req, res, next) => {
     const fields = ['certificateNumber', 'accreditingBody', 'passDate', 'modules'];
     const values = pick(req.session.form[req.model.id].values, fields);
 
-    values.modules = values.modules.map(module => ({ module, species: [] }));
+    values.modules = values.modules.map(module => {
+      const species = req.form.values[`module-${module}-species`];
+      return { module,
+        species:
+         Array.isArray(species) ? species.filter(s => s !== '') : (species !== '' ? [species] : [])
```
Comment (joefitter) [inline pages/pil/training/routers/modules.js]: indentation

Comment (joefitter) [reply pages/pil/training/routers/modules.js]: also, what are you trying to do here? It looks a lot like you want `castArray` or something

File: pages/common/views/components/add-another.jsx (hunk 1 above)
Comment (lennym) [reply pages/common/views/components/add-another.jsx]: Nah, `e.preventDefault()` will have caught that.

File: pages/pil/training/routers/modules.js [hunk 3]
```diff
@@ -14,13 +14,35 @@ module.exports = settings => {
...
+      };
+    }
+    );
```
Comment (joefitter) [inline pages/pil/training/routers/modules.js]: can we bring this up a line, `});`

File: pages/pil/training/views/modules.jsx [hunk 4]
```diff
@@ -1,9 +1,39 @@
 import React from 'react';
 import Snippet from '../../../common/views/containers/snippet';
 import FormLayout from '../../../common/views/layouts/form';
+import Inset from '../../../common/views/components/inset';
+import AddAnother from '../../../common/views/components/add-another';
+import { Select } from '@ukhomeoffice/react-components';
+const { species } = require('@asl/constants');
```
Comment (joefitter) [inline pages/pil/training/views/modules.jsx]: mixed import and require

File: pages/pil/training/routers/modules.js (hunk 3 above)
Comment (lennym) [reply pages/pil/training/routers/modules.js]: This is some pretty wild code. Given that we always want to end up with an array, I'd do that casting first, and then filter afterwards.

i.e. `[].concat(species).filter(Boolean)` will achieve the same thing.

File: pages/pil/training/routers/modules.js (hunk 2 above)
Comment (lennym) [reply pages/pil/training/routers/modules.js]: See my other comment 👇

**Comment by joefitter** [comment]
@lvnhmd can you please not self merge unless a PR has at lease been approved?

**Pull request merged**